tests/models/test_lightfm.py

```diff
@@ -222,6 +222,140 @@ def test_with_weights(self, interactions_df: pd.DataFrame) -> None:
+        )
+
+    def test_fit_partial_with_features(self, dataset_with_features: Dataset) -> None:
+        base_model = DeterministicLightFM(no_components=2, loss="logistic")
+        model = LightFMWrapperModel(model=base_model, epochs=50).fit(dataset_with_features)
+        model.fit_partial(dataset_with_features)
+        actual = model.recommend(
+            users=np.array([10, 150]),  # new user
+            dataset=dataset_with_features,
+            k=2,
+            filter_viewed=False,
+        )
+        expected = pd.DataFrame(
+            {
+                Columns.User: [10, 10, 150, 150],
+                Columns.Item: [11, 12, 11, 12],
+                Columns.Rank: [1, 2, 1, 2],
+            }
+        )
+        pd.testing.assert_frame_equal(actual.drop(columns=Columns.Score), expected)
+        pd.testing.assert_frame_equal(
+            actual.sort_values([Columns.User, Columns.Score], ascending=[True, False]).reset_index(drop=True),
+            actual,
+        )
+
+    def test_fit_partial_without_fit(self, dataset: Dataset) -> None:
+        base_model = DeterministicLightFM(no_components=2, loss="logistic")
+        model = LightFMWrapperModel(model=base_model, epochs=50).fit_partial(dataset)
+        actual = model.recommend(
+            users=np.array([10, 20, 150]),  # hot, hot, cold
+            dataset=dataset,
+            k=2,
+            filter_viewed=False,
+        )
+        expected = pd.DataFrame(
+            {
+                Columns.User: [10, 10, 20, 20, 150, 150],
+                Columns.Item: [11, 12, 11, 12, 11, 12],
+                Columns.Rank: [1, 2, 1, 2, 1, 2],
+            }
+        )
+        pd.testing.assert_frame_equal(actual.drop(columns=Columns.Score), expected)
+        pd.testing.assert_frame_equal(
+            actual.sort_values([Columns.User, Columns.Score], ascending=[True, False]).reset_index(drop=True),
+            actual,
+        )
+
+    def test_fit_partial_twice(self, dataset: Dataset) -> None:
+        base_model = DeterministicLightFM(no_components=2, loss="logistic")
+        model = LightFMWrapperModel(model=base_model, epochs=50).fit_partial(dataset)
+        model.fit_partial(dataset)
+        actual = model.recommend(
+            users=np.array([10, 20, 150]),  # hot, hot, cold
+            dataset=dataset,
+            k=2,
+            filter_viewed=False,
+        )
+        expected = pd.DataFrame(
+            {
+                Columns.User: [10, 10, 20, 20, 150, 150],
+                Columns.Item: [11, 12, 11, 12, 11, 12],
+                Columns.Rank: [1, 2, 1, 2, 1, 2],
+            }
+        )
+        pd.testing.assert_frame_equal(actual.drop(columns=Columns.Score), expected)
+        pd.testing.assert_frame_equal(
+            actual.sort_values([Columns.User, Columns.Score], ascending=[True, False]).reset_index(drop=True),
+            actual,
+        )
+
+    def test_fit_partial_with_same_dataset(self, dataset: Dataset) -> None:
```

Comment:
Please make this a test where factors are being compared. Same recommendations on small toy data  are not really a signal that `fit_partial` and `fit` are doing exactly the same logic. Please `fit` model for 10 epochs. Then `fit_partial` 10 times for one epoch on another copy of model. And then please compare that factors are the same